Source/Core/VideoCommon/OnScreenUI.cpp

```diff
@@ -326,13 +327,78 @@ void OnScreenUI::DrawDebugText()
     ImGui::TextUnformatted(profile_output.c_str());
 }
 
+#ifdef USE_RETRO_ACHIEVEMENTS
+void OnScreenUI::DrawChallenges()
+{
+  std::lock_guard lg{*AchievementManager::GetInstance()->GetLock()};
+  const AchievementManager::NamedIconMap& challenge_icons =
+      AchievementManager::GetInstance()->GetChallengeIcons();
+
+  const std::string window_name = "Challenges";
+
+  u32 sum_of_icon_heights = 0;
+  u32 max_icon_width = 0;
+  for (const auto& [name, icon] : challenge_icons)
+  {
+    // These *should* all be the same square size but you never know.
+    if (icon->width > max_icon_width)
+      max_icon_width = icon->width;
+    sum_of_icon_heights += icon->height;
+  }
+  ImGui::SetNextWindowPos(
+      ImVec2(ImGui::GetIO().DisplaySize.x - 20.f * m_backbuffer_scale - max_icon_width,
+             ImGui::GetIO().DisplaySize.y - 20.f * m_backbuffer_scale - sum_of_icon_heights));
+  ImGui::SetNextWindowSize(ImVec2(0.0f, 0.0f));
+  if (ImGui::Begin(window_name.c_str(), nullptr,
+                   ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoInputs |
+                       ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoSavedSettings |
+                       ImGuiWindowFlags_NoScrollbar | ImGuiWindowFlags_NoNav |
+                       ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoFocusOnAppearing))
+  {
+    for (const auto& [name, icon] : challenge_icons)
+    {
+      if (m_challenge_texture_map.find(name) != m_challenge_texture_map.end())
+        continue;
+      const u32 width = icon->width;
+      const u32 height = icon->height;
+      TextureConfig tex_config(width, height, 1, 1, 1, AbstractTextureFormat::RGBA8, 0);
+      m_challenge_texture_map[name] = g_gfx->CreateTexture(tex_config);
```

Comment:
```suggestion
      auto res = m_challenge_texture_map,insert_or_assign(name, g_gfx->CreateTexture(tex_config));
```

then use `res` for the if statement and below, to avoid doing two redundant map lookups